dashboard/app/controllers/concerns/pd/workshop_filters.rb

```diff
@@ -83,11 +83,17 @@ def filter_workshops(workshops)
       workshops = workshops.scheduled_start_on_or_before(ensure_date(params[:end])) if params[:end]
       workshops = workshops.where(course: params[:course]) if params[:course]
       workshops = workshops.where(subject: params[:subject]) if params[:subject]
-      workshops = workshops.where(organizer_id: params[:organizer_id]) if params[:organizer_id]
+
+      if current_user.permission?(UserPermission::WORKSHOP_ADMIN)
+        workshops = workshops.where(organizer_id: params[:organizer_id]) if params[:organizer_id]
+        workshops = workshops.facilitated_by(User.find(params[:facilitator_id])) if params[:facilitator_id]
```

Comment:
What if we pass a bad `facilitator_id`? I'd expect zero results, or maybe a bad request, but this `find` will raise an `ActiveRecord::RecordNotFound` which will cause a 404 (not found) response. It might be nicer to use `find_by(id: ` which will not raise the error. Not a big deal though, your call.